### PR TITLE
add faster implementation for `findall(::AbstractVector{Bool})`

### PR DESCRIPTION
**Pull request by mzgubic**
```julia
julia> a = rand(Bool, 10000);

julia> @btime findall($a);
  40.073 μs (2 allocations: 38.95 KiB)

julia> function myfindall(a)
           return (1:length(a))[a]
       end

julia> @btime myfindall($a);
  33.978 μs (2 allocations: 38.95 KiB)
```

though seems to be somewhat slower with BitArrays

```julia
julia> b = BitArray(rand(Bool, 10_000));

julia> @btime findall($b);
  6.113 μs (2 allocations: 39.89 KiB)

julia> @btime myfindall($b);
  6.224 μs (2 allocations: 39.89 KiB)
```

Thanks to @eperim  for finding this

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -2366,6 +2366,7 @@ function findall(A::AbstractArray{Bool})
     end
     I
 end
+findall(A::AbstractVector{Bool}) = (1:length(A))[A]
```
Comment (oxinabox) [inline base/array.jl:2369]: To make sure e.g. OffsetArrays don't break:
```suggestion
findall(A::AbstractVector{Bool}) = eachindex(A)[A]
```

**Comment by jakobnissen** [comment]
There is already: https://github.com/JuliaLang/julia/pull/42202

**Comment by mzgubic** [comment]
Oh, nice! Is your approach even faster for the benchmarks above?

**Comment by jakobnissen** [comment]
Yes - on my branch:
```
julia> a = rand(Bool, 10000);

julia> using BenchmarkTools

julia> @btime findall($a);
  5.085 μs (2 allocations: 38.92 KiB)

julia> function myfindall(a)
                  return (1:length(a))[a]
              end
myfindall (generic function with 1 method)

julia> @btime myfindall($a);
  29.697 μs (2 allocations: 38.92 KiB)
```

**Comment by mzgubic** [comment]
Excellent!

**Pull request closed without merging**